internal/pkg/template/diff/override.go

```diff
@@ -34,18 +36,65 @@ func (m *ignorer) match(_, _ *yaml.Node, key string) bool {
 }
 
 // Parse is a no-op for an ignorer.
-func (m *ignorer) parse(_, _ *yaml.Node, _ string) (diffNode, error) {
+func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error) {
 	return nil, nil
 }
 
-type noopOverrider struct{}
+// intrinsicFuncFullShortFormConverter handles comparison between full/short form of an intrinsic function.
+type intrinsicFuncFullShortFormConverter struct{}
 
-// Match always returns false for a noopOverrider.
-func (*noopOverrider) match(_, _ *yaml.Node, _ string) bool {
-	return false
+// match returns true if from and to node represent the same intrinsic function written in different (full/short) form.
+// Example1: "!Ref abc" and "Ref: abc" will return true.
```

Comment:
huh this made me think that it's strange CFN doesn't do `Fn::Ref` instead of `Ref`.

Like what if a service API's property name was called `Ref` 🤯 what would happen lol

Reply:
nit: do you think we should update the comment to not include any values like `!Ref` and `Ref:` as examples rather than `!Ref abc`